.github/drafts/16521/Code-scanner-Draft.md

```diff
@@ -0,0 +1,122 @@
+| [Animate Loading](https://docs.appsmith.com/reference/widgets)           | Formatting           | It sets a tooltip for the widget. You can add hints or extra information about the required input from the user. Learn more about [Tooltip](https://docs.appsmith.com/reference/widgets)    | -                           |
+| value           | Binding              | Fetches the value of scanned code                                                                                   | {{CodeScanner1.value}}      |
+|                 |                      |                                                                                                                     |                             |
+
+
+
+### Text
+
+Text/Label is a widget property that lets you set the text inside the button, describing the function it performs. It's used to display static text on the button. For example, you can enter "Scan Menu" as the label for a button that scans the restaurant menu.
+
+[Text Video](https://youtu.be/o80-IKcXAVQ)
+
+
+### Value
+
+Value is a binding property, it fetches the response from the scanned code. To bind the value of the code scanner widget to another widget, open the property pane of it, and add the code snippet given below:
+
+```
+{{<your_widget>.value}}
+```
+
+
+For example, let's take a Code Scanner widget(CodeScanner1) and bind its value to a text widget. Drag a text widget onto the canvas and add the following code:
+
+```
+{{CodeScanner1.value}}
+```
+
+.https://youtu.be/zfkpIzaiTX0
+
+
+# Events
+
+They are a set of actions that you can perform on the widget. The following table lists the actions:
+
+| Event          |  Description                                     |                                            
+|----------------|--------------------------------------------------|
+| onCodeDetected | Triggers an action when a valid code is detected. See a list of [supported actions](https://docs.appsmith.com/reference/appsmith-framework/widget-actions).|
+
+### onCodeDetected 
+
+When a valid barcode or QR code is detected, this event triggers an action. 
+The Appsmith framework allows triggering actions for widget events and inside JS Objects. Let’s take an example to understand how the onCodeDetected event works.
+
+  
+In this example, we will use the [Modal widget](https://docs.appsmith.com/reference/widgets/modal) to display the code response.
+
+* First, set the onCodeDetected event to open a New Modal, and choose to Create New.
+* This will open up a new modal; let's drag the [text widget](https://docs.appsmith.com/reference/widgets/text) into the modal.
+* In the property section, add the following code:
+  
+```
+{{CodeScanner1.value}}
+```
+
+With this, whenever a valid code is detected, a modal window displaying the code response will show up. The video below demonstrates how to achieve that.
+  
+https://youtu.be/_2p1bMbdk6U
+
+
+# Styles
```

Comment:
We should rename this to **Styling** (also in the widget doc template) . thoughts @Pranay105 @jnikhila